docs/reference/technology/firmware_update.md

```diff
@@ -21,31 +22,34 @@ The active firmware, made up of Mbed OS and an application, can be updated only
 
-To protect against faults in the newly updated components, we store multiple versions (up to a certain maximum number of versions). Each stag is responsible for detecting faults in the following component and rolling it back if it discovers faults. For example, if stage 3 is unstable, needs additional functionality or is otherwise behaving incorrectly, stage 2 (during the *next* startup sequence) can roll stage 3 back to an earlier version before handing over control. This results in a boot-sequence tree that is traversed in a depth-first order as the system recovers from successive faults.
+To protect against faults in the newly updated components, we store multiple versions (up to a certain maximum number of versions). <!---What is that number?---> Each stage is responsible for detecting faults in the following component and rolling it back if it discovers faults. For example, if the third stage is unstable, needs additional functionality or is otherwise behaving incorrectly, the second stage (during the *next* startup sequence) can roll the third stage back to an earlier version before handing over control. This results in a boot sequence tree that is traversed in a depth-first order as the system recovers from successive faults.
+>That is traversed?
 
-Most boot sequences are composed of only three stages:
+Most boot sequences are composed of three stages:
 
 1. Boot selector (also known as **root bootloader** or **stage zero bootloader**): does not get upgraded.
 1. Bootloader: upgradable, with several versions stored on the device.
 1. Application: upgradable, with several versions stored on the device.
 
 Fault tolerance ultimately rests on the sanity of the first-stage bootloader. This bootloader is usually kept minimal to ensure dependable operation.
 
-The Mbed OS bootloader is a hybrid of the boot selector and a bootloader, but it fulfills the requirements of the boot selector: It is small enough to minimize the chance of bugs, but it is advanced enough to install new images.
+The Mbed OS bootloader is a hybrid of the boot selector and a bootloader, but it fulfills the requirements of the boot selector: It is small enough to minimize the chance of bugs, but advanced enough to install new images.
 
 ### Managed and unmanaged bootloader tool integration
 
-The Mbed tools (Mbed CLI, Online Compiler) can manage bootloaders where:
+Mbed tools (Mbed CLI, Online Compiler) can manage bootloaders where:
 
 - The bootloader comes before the application in ROM.
 - The application starts immediately after the bootloader.
 
 If the Mbed tool finds a manageable bootloader, the image build process automatically merges the bootloader image with the application image.
+>We say managed and manageable--which one should we go with?
```

Comment:
"Managed." We have "managed" and "unmanaged." "Manageable" and "unmanageable" would have very different connotations :)